File: 366_fibonacci/fibonacci.java
/*
@Copyright:LintCode
@Author:   shaman
@Problem:  http://www.lintcode.com/problem/fibonacci
@Language: Java
@Datetime: 16-01-28 15:18
*/

public class Solution {
    public int fibonacci(int n) {
        if (n == 1 || n == 2) {
            return n - 1;
        }

        return fibHelper(0, 1, n);
    }

    private int fibHelper(int i, int j, int n) {
        if ( n == 2) {
            return j;
        }
        return fibHelper(j, i + j, n - 1);
    }
}

File: 366_fibonacci/fibonacci.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   shaman
@Problem:  http://www.lintcode.com/problem/fibonacci
@Language: Python
@Datetime: 16-01-30 12:44
'''

class Solution:
    # @param n: an integer
    # @return an integer f(n)
    def fibonacci(self, n):
        # write your code here
        a, b = 0, 1
        for i in range(1, n):
            a, b = b, a + b
        return a